From 5832b4d0d76bd0819b772a98c39f012c2088f227 Mon Sep 17 00:00:00 2001
From: Patrick Delaunay <patrick.delaunay@foss.st.com>
Date: Fri, 8 Dec 2023 13:10:32 +0100
Subject: [PATCH 0571/1695] dts: stm32: force v5v_vconn always on for
 stm32mp257f-dk

Workaround to allow M33 firmware development for USB UCPD

Need to be reverted with support of
* Ticket 147183 - MP25, add SCMI client in CM33 FW, USB UCPD need SCMI
  service to enable its power supplies

Upstream-Status: Inappropriate [ST downstream https://github.com/STMicroelectronics/optee_os.git eb0ce0780ac676650cbc46f2d18b85f260a7b0ff dts: stm32: force v5v_vconn always on for stm32mp257f-dk]
Change-Id: Iba3d3b8c3d575cd98d7b622e02f911261342ef08
Signed-off-by: Patrick Delaunay <patrick.delaunay@foss.st.com>
Reviewed-on: https://gerrit.st.com/c/mpu/oe/optee/optee_os/+/344912
ACI: CIBUILD <MDG-smet-aci-builds@list.st.com>
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
Reviewed-by: Gatien CHEVALLIER <gatien.chevallier@st.com>
---
 core/arch/arm/dts/stm32mp257f-dk.dts | 1 +
 1 file changed, 1 insertion(+)

diff --git a/core/arch/arm/dts/stm32mp257f-dk.dts b/core/arch/arm/dts/stm32mp257f-dk.dts
index 019d9832a..b8ee298d2 100644
--- a/core/arch/arm/dts/stm32mp257f-dk.dts
+++ b/core/arch/arm/dts/stm32mp257f-dk.dts
@@ -271,6 +271,7 @@
 				regulator-name = "v5v_vconn";
 				regulator-min-microvolt = <4000000>;
 				regulator-max-microvolt = <4000000>;
+				regulator-always-on;
 				st,regulator-bypass-microvolt = <4000000>;
 				st,pwrctrl_sel = <0>;
 				st,pwrctrl_reset;
-- 
2.47.2

